src/api/cvc4cpp.cpp

```diff
@@ -4263,34 +4264,43 @@ Term Solver::defineFun(Term fun,
                        Term term) const
 {
   CVC4_API_SOLVER_TRY_CATCH_BEGIN;
-  CVC4_API_ARG_CHECK_EXPECTED(fun.getSort().isFunction(), fun) << "function";
-  std::vector<Sort> domain_sorts = fun.getSort().getFunctionDomainSorts();
-  size_t size = bound_vars.size();
-  CVC4_API_ARG_SIZE_CHECK_EXPECTED(size == domain_sorts.size(), bound_vars)
-      << "'" << domain_sorts.size() << "'";
-  for (size_t i = 0; i < size; ++i)
+
+  if (fun.getSort().isFunction())
```

Comment:
```suggestion
  CVC4_API_ARG_CHECK_EXPECTED(fun.getSort().isFunction() || bound_vars.size() == 0, fun)
        << "function or nullary symbol";
```
Can't this simplified like this? (The else case below would of course be deleted then.)

Reply:
Actually, that's what I had at first but it didn't work out. The reason is that `fun.getSort().getFunctionDomainSorts()` throws an error when it is called on a non-function sort.

Reply:
Oh, I see. Alright.